docs/features/software-catalog/descriptor-format.md

```diff
@@ -0,0 +1,295 @@
+for a parser to know how to interpret the contents of the rest of the data.
+
+Backstage specific entities have an `apiVersion` that is prefixed with
+`backstage.io/`, to distinguish them from other types of object that share
+the same type of structure. This may be relevant when co-hosting these
+specifications with e.g. kubernetes object manifests, or when an organization
+adds their own specific kinds of entity to the catalog.
+
+Early versions of the catalog will be using beta versions, e.g. `backstage.io/v1beta1`,
+to signal that the format may still change. After that, we will be using
+`backstage.io/v1` and up.
+
+### `metadata` [required]
+
+A structure that contains metadata about the entity, i.e. things that aren't directly
+part of the entity specification itself. See below for more details about this structure.
+
+### `spec` [varies]
+
+The actual specification data that describes the entity.
+
+The precise structure of the `spec` depends on the `apiVersion` and `kind` combination,
+and some kinds may not even have a `spec` at all. See further down in this document for
+the specification structure of specific kinds.
+
+## Common to All Kinds: The Metadata
+
+The `metadata` root field has a number of reserved fields with specific meaning, described
+below.
+
+In addition to these, you may add any number of other fields directly under `metadata`, but
+be aware that general plugins and tools may not be able to understand their semantics.
+
+### `name` [required]
+
+The name of the entity. This name is both meant for human eyes to recognize the entity,
+and for machines and other components to reference the entity (e.g. in URLs or from
+other entity specification files).
+
+Names must be unique per kind, within a given namespace (if specified), at any point in
+time. Names may be reused at a later time, after an entity is deleted from the registry.
+
+Names are required to follow a certain format. Entities that do not follow those rules
+will not be accepted for registration in the catalog. The ruleset is configurable to fit
+your organization's needs, but the default behavior is as follows.
+
+- Strings of length at least 1, and at most 63
+- Must consist of sequences of `[a-z0-9A-Z]` possibly separated by one of `[-_.]`
+
+Example: `visits-tracking-service`, `CircleciBuildsDump_avro_gcs`
+
+In addition to this, names are passed through a normalization function and then compared
+to the same normalized form of other entity names and made sure to not collide. This rule
+of uniqueness exists to avoid situations where e.g. both `my-component` and `MyComponent`
+are registered side by side, which leads to confusion and risk. The normalization function
+is also configurable, but the default behavior is as follows.
+
+- Strip out all characters outside of the set `[a-zA-Z0-9]`
+- Convert to lowercase
+
+Example: `CircleciBuildsDs_avro_gcs` -> `circlecibuildsdsavrogcs`
```

Comment:
Wouldn't this be CircleciBuildsDsavrogcs if all but `[a-zA-Z0-9]` is stripped? (Genuine question since my regex-fu is bad.

Reply:
Note that it also says "convert to lowercase" after that :)

Reply:
Ohh, sorry. It all checks out then. :)